odoo/addons/base/models/qweb.py

```diff
@@ -864,832 +625,506 @@ def _compile_static_node(self, el, options):
+        else:
+            code.append(self._indent("force_display = None", indent))
 
-    def _compile_directive_foreach(self, el, options):
-        expr = self._compile_expr(el.attrib.pop('t-foreach'))
-        varname = el.attrib.pop('t-as').replace('.', '_')
-        values = self._make_name('values')
-
-        # create function $foreach
-        def_name = self._create_def(options, self._compile_directives(el, options), prefix='foreach', lineno=el.sourceline)
-
-        # for $values in foreach_iterator(values, $expr, $varname):
-        #     $foreach(self, append, $values, options)
-        return [ast.For(
-            target=ast.Name(id=values, ctx=ast.Store()),
-            iter=ast.Call(
-                func=ast.Name(id='foreach_iterator', ctx=ast.Load()),
-                args=[ast.Name(id='values', ctx=ast.Load()), expr, ast.Str(varname)],
-                keywords=[], starargs=None, kwargs=None
-            ),
-            body=[ast.Expr(self._call_def(def_name, values=values))],
-            orelse=[]
-        )]
-
-    def _compile_tail(self, el):
-        return el.tail is not None and [self._append(ast.Str(pycompat.to_text(el.tail)))] or []
-
-    def _compile_directive_esc(self, el, options):
+        code.extend(self._compile_widget_value(el, options, indent, without_attributes=not code_options))
+        return code
+
+    def _compile_directive_esc(self, el, options, indent):
         el.attrib['t-out'] = el.attrib.pop('t-esc')
-        return self._compile_directive_out(el, options)
-    def _compile_directive_out(self, el, options):
-        field_options = self._compile_widget_options(el)
-        content = self._compile_widget(el, el.attrib.pop('t-out'), field_options)
-        if not field_options:
-            # if content is not False and if content is not None:
-            #     content = escape(pycompat.to_text(content))
-            content.append(self._if_content_is_not_Falsy([
-                ast.Assign(
-                    targets=[ast.Name(id='content', ctx=ast.Store())],
-                    value=ast.Call(
-                        func=ast.Name(id='escape', ctx=ast.Load()),
-                        args=[ast.Call(
-                            func=ast.Name(id='to_text', ctx=ast.Load()),
-                            args=[ast.Name(id='content', ctx=ast.Load())], keywords=[],
-                            starargs=None, kwargs=None
-                        )],
-                        keywords=[],
-                        starargs=None, kwargs=None
-                    )
-                )],
-                []
-            ))
-        return content + self._compile_widget_value(el, options)
-
-    def _compile_directive_raw(self, el, options):
+        if not 't-options' in el.attrib and not any(key for key in el.attrib if key.startswith('t-options-')):
+            el.attrib['t-options'] = '{"widget": "esc"}'
+        return self._compile_directive_out(el, options, indent)
```

Comment:
`t-esc` being a (soft-deprecated) alias for `t-out` was intentional.

Reply:
In case you want to force the escaping:

```xml
<t t-set="o">
  <script>toto = 5;</script>
</t>
...
<t t-esc="o"/>
```
Result:
```xml
  &lt;script&gt;toto = 5;&lt;/script&gt;
```

Reply:
The recommended method then is to `str()` the content to strip out the Markup-ness.

Reply:
To force escaping escape you have to write in the template `t-out="escape(str(content))"` still do I put a log as for t-raw with a message so the user knows what to do?